ci/playbooks/pre-doc.yml

```diff
@@ -23,5 +23,19 @@
         name:
           - make
           - python3
-          - golang
           - ruby
+
+    - name: Download golang version 1.21.7
+      ansible.builtin.get_url:
+        source: https://go.dev/dl/go1.21.7.linux-amd64.tar.gz
+        dest: go1.21.7.linux-amd64.tar.gz
+        checksum: ffeabdfe10a7e84197ac7d28fda7484d0f855a92
+
+    - name: Install golang
+      become: true
+      ansible.builtin.command:
+        cmd: |
+          tar -C ~/bin -xzf go1.21.7.linux-amd64.tar.gz
+          export PATH=$PATH:~/bin/go/bin
+        args:
+          creates: /usr/local/bin/go
```

Comment:
```suggestion
    - name: Install golang
      ansible.builtin.unarchive:
        src: go1.21.7.linux-amd64.tar.gz
        dest: ~/bin/
        remote_src: true
        include: go/bin/
        extra_opts:
          - --strip-components=2
```

Reply:
we could also use the unarchive module to download the tar, but we'd lose the checksum verification.